org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLCompletions.java

```diff
@@ -607,6 +610,24 @@ private void collectAttributeNameSuggestions(int nameStart, int nameEnd, Complet
 		}
 	}
 
+	private boolean computeXSIAttribute(int nameStart, int nameEnd, String text, CompletionRequest request, CompletionResponse response) {
+		Range editRange = null;
+		try {
+			editRange = getReplaceRange(nameStart, nameEnd, request);
+		} catch (BadLocationException e) {
+			LOGGER.log(Level.SEVERE,"BadLocation in isXSIAttribute() while trying to get completion items",e);
+		}
+		String name = text.substring(nameStart, nameEnd);
+		XSISchemaModel model = new XSISchemaModel(name, editRange, response, request);

```

Comment:
IMHO, XSISchemaModel  should not exists or should not be an instance. You could have your methods which waits ICompletionRequest which provides all fields from the model (range, etc). Please use XSISchemaModel  with static method (utility methods).